### PR TITLE
Add wrapping methods for java.uti.Map

### PR DESCRIPTION
**Pull request by v1ctor**
Hi! 
I've added useful methods for wrapping standard java.util.Map realizations to javaslang Map types.

### CODE REVIEW

**Comment by v1ctor** [comment]
I've passed all the tests on my own machine and it's have finished successfully.
How can I reprocess the tests at Travis CI?

**Comment by danieldietrich** [comment]
I restarted the travis-ci test. I think you have to be admin to do that. It seems there was a timeout in a wait-method of a Future-unit test. I will increase the wait time-window a bit then it should not happen again...

File: src/main/java/javaslang/collection/HashMap.java [hunk 1]
```diff
@@ -78,6 +78,19 @@ private HashMap(HashArrayMappedTrie<K, V> trie) {
...
     /**
+     * Returns a {@code HashMap}, from a source java.util.Map.
+     *
+     * @param map A map entry.
+     * @param <K>   The key type
+     * @param <V>   The value type
+     * @return A new Map containing the given map
+     */
+    @SuppressWarnings("unchecked")
+    public static <K, V> HashMap<K, V> of(java.util.Map<? extends K, ? extends V> map) {
```
Comment (danieldietrich) [inline src/main/java/javaslang/collection/HashMap.java]: We need to name it `ofAll`

File: src/main/java/javaslang/collection/HashMap.java [hunk 2]
```diff
@@ -389,6 +405,16 @@ public boolean isTraversableAgain() {
     }
 
     @Override
+    public HashMap<K, V> put(java.util.Map<? extends K, ? extends V> map) {
```
Comment (danieldietrich) [inline src/main/java/javaslang/collection/HashMap.java]: It is not a good idea to have this put method. In fact it is a _merge_ of maps which needs to handle collisions. We already have it.

Also I do not want to add specific java.util.* methods to the Javaslang library. For conversion we have the `ofAll` factory methods and the `toJavaXxx()` methods.

Please move the body of the `put` methods to the `ofAll(java.util.Map)` methods. We then can do the following:

```java
java.util.Map<...> map1 = ...;
javaslang.collection.Map<...> map2 = ...;
map2.merge(HashMap.ofAll(map1));
```

File: src/main/java/javaslang/collection/LinkedHashMap.java [hunk 3]
```diff
@@ -78,6 +78,19 @@ public LinkedHashMap(Queue<Tuple2<K, V>> list, HashMap<K, V> map) {
...
     /**
+     * Returns a {@code LinkedHashMap}, from a source java.util.Map.
+     *
+     * @param map A map entry.
+     * @param <K>   The key type
+     * @param <V>   The value type
+     * @return A new Map containing the given map
+     */
+    @SuppressWarnings("unchecked")
+    public static <K, V> LinkedHashMap<K, V> of(java.util.Map<? extends K, ? extends V> map) {
```
Comment (danieldietrich) [inline src/main/java/javaslang/collection/LinkedHashMap.java]: also `ofAll`

File: src/main/java/javaslang/collection/LinkedHashMap.java [hunk 4]
```diff
@@ -193,6 +206,16 @@ public boolean containsValue(V value) {
     }
 
     @Override
+    public LinkedHashMap<K, V> put(java.util.Map<? extends K, ? extends V> map) {
```
Comment (danieldietrich) [inline src/main/java/javaslang/collection/LinkedHashMap.java]: also move to `ofAll` and use `merge` instead like described above

File: src/main/java/javaslang/collection/Map.java [hunk 5]
```diff
@@ -59,6 +59,14 @@ default V apply(K key) {
      */
     Map<K, V> put(Tuple2<? extends K, ? extends V> entry);
 
+    /**
+     * Convenience method for {@code put(map)}.
+     *
+     * @param map java.util.Map
+     * @return A new Map containing all elements of the source map.
+     */
+    Map<K, V> put(java.util.Map<? extends K, ? extends V> map);
```
Comment (danieldietrich) [inline src/main/java/javaslang/collection/Map.java]: can be removed

File: src/main/java/javaslang/collection/TreeMap.java [hunk 6]
```diff
@@ -99,6 +99,19 @@ private TreeMap(RedBlackTree<Tuple2<K, V>> entries) {
...
     /**
+     * Returns a {@code TreeMap}, from a source java.util.Map.
+     *
+     * @param map A map entry.
+     * @param <K>   The key type
+     * @param <V>   The value type
+     * @return A new Map containing the given map
+     */
+    @SuppressWarnings("unchecked")
+    public static <K extends Comparable<? super K>, V> TreeMap<K, V> of(java.util.Map<? extends K, ? extends V> map) {
```
Comment (danieldietrich) [inline src/main/java/javaslang/collection/TreeMap.java]: `ofAll`

File: src/main/java/javaslang/collection/TreeMap.java [hunk 7]
```diff
@@ -490,6 +503,16 @@ public boolean isTraversableAgain() {
     }
 
     @Override
+    public TreeMap<K, V> put(java.util.Map<? extends K, ? extends V> map) {
```
Comment (danieldietrich) [inline src/main/java/javaslang/collection/TreeMap.java]: see above

File: src/test/java/javaslang/collection/AbstractMapTest.java [hunk 8]
```diff
@@ -330,6 +330,14 @@ public void shouldPutTuple() {
         assertThat(emptyIntInt().put(Tuple.of(1, 2))).isEqualTo(emptyIntInt().put(1, 2));
     }
 
+    @Test
+    public void shouldPutMap() {
```
Comment (danieldietrich) [inline src/test/java/javaslang/collection/AbstractMapTest.java]: shouldMergeJavaUtilMap

**Comment by v1ctor** [comment]
Thanks for you marks, I've fixed all the mistakes :)

**Comment by danieldietrich** [comment]
Don't get why it does not build on travis-ci. Will investigate it tomorrow morning!

**Comment by danieldietrich** [comment]
I will pull it. The error has nothing to do with the code changed here. Afterwards I will fix it with a separate PR.

Many thx!

**Comment by v1ctor** [comment]
Thanks!

**Pull request merged**